src/casa.wlk

```diff
@@ -1,3 +1,134 @@
+//CASA
 
 object casaDePepeYJulian {
+	
+	var property viveres = 50
+	var property reparaciones = 0
+	var property cuenta = cuentaCorriente
+	var property estrategiaDeAhorro = minimoEIndispensable
+	
+	method tieneViveresSuficientes(){
+		return viveres > 40
+	}
+	
+	method hayQueHacerReparaciones(){
+		return reparaciones > 0
+	}
+	
+	method laCasaEstaEnOrden(){
+		return not self.hayQueHacerReparaciones() and self.tieneViveresSuficientes()
+	}
+	
+	method romperAlgo(costoReparaciones){
+		reparaciones += costoReparaciones
+	}
+	
+	method reparar(){
+		cuenta.extraer(reparaciones)
+		reparaciones = 0
+	}
+	
+	method mantener(){
+		estrategiaDeAhorro.mantener(self)
+	}
+	
+	method comprarViveres(porcentaje, calidad){
+		viveres += porcentaje
+		cuenta.extraer(porcentaje * calidad)
+	}
+	
+	method puedePagarReparaciones(){
+		return cuenta.saldo() >= reparaciones
+	}
+	
+	method sobranteAlReparar(){
+		return cuenta.saldo() - reparaciones
+	}
+	
 }
+
+//CUENTAS
+
+object cuentaCorriente {
+	
+	var property saldo = 0
+	
+	method depositar(cantidad){
+		saldo += cantidad
+	}
+	
+	method extraer(cantidad){
+		saldo -= cantidad
+	}
+	
+}
+
+object cuentaConGastos {
+	
+	var property saldo = 0
+	var property costoPorOperacion = 0
+	
+	method depositar(cantidad){
+		saldo += cantidad - costoPorOperacion
+	}
+	
+	method extraer(cantidad){
+		saldo -= cantidad
+	}
+	
+}
+
+object cuentaCombinada {
+	
+	var property cuentaPrimaria = cuentaConGastos
+	var property cuentaSecundaria = cuentaCorriente
+	
+	method saldo(){
+		return cuentaPrimaria.saldo() + cuentaSecundaria.saldo()
+	}
+	
+	method depositar(cantidad){
+		cuentaPrimaria.depositar(cantidad)
+	}
+	
+	method extraer(cantidad){
+		if(cuentaPrimaria.saldo() > cantidad){
+			cuentaPrimaria.extraer(cantidad)
+		} else {
+			cuentaSecundaria.extraer(cantidad)
+		}
+	}
+	
+}
+
+//ESTRATEGIAS DE AHORRO
+
+object minimoEIndispensable {
+	
+	const calidad = 5
+	
+	method mantener(casa){
+		if(not casa.tieneViveresSuficientes()){
+			const viveresFaltantes = 40 - casa.viveres()
```

Comment:
Ese 40 significa que es el mínimo para considerar que la casa está en orden. Quedó duplicado con la lógica de la casa, lo ideal sería preguntarle a la casa

Reply:
Profe, no me deja pushear la corrección, me dice: "Can't connect to any repository: https://github.com/obj1unq/2024s1---casa-de-pepe-y-julian-Lautaro-Blanco.git (https://github.com/obj1unq/2024s1---casa-de-pepe-y-julian-Lautaro-Blanco.git: not authorized)"

Lo que le corregí es el if: "method mantener(casa){
		if(casa.viveres() > 40){
			const viveresFaltantes = 40 - casa.viveres()
			casa.comprarViveres(viveresFaltantes, calidad)
		}
	}"